### ISSUE TITLE
Node Red 3.0.0 flows not displaying in web ui

### ISSUE THREAD

**Issue by Den-W1** [hassio-addons/addon-node-red#197]
# Problem/Motivation

> (Why the issue was filed)

## Expected behavior

> (What you expected to happen)

## Actual behavior

> (What actually happened)

## Steps to reproduce

> (How can someone else make/see it happen)

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

**Comment by Den-W1** [comment]
Node Red ingress flows not displaying after upgrade to Node Red 3.0.0
After update open web ui displays sometimes.  Restart, Reboot close and restart browser no change.
Sometimes = Just after install web ui did display all flows.  However after closing and then returning it did not.  

Node Red 3.0.0
Home Assistant (Hass.io) 0.91.4   Via Proxmox
Google Chrome Version 73.0.3683.103 (Official Build) (64-bit)

**Comment by SurfingSpider** [comment]
I get the same thing. I have just updated again to the latest version of home assistant and node-red I still dont get the UI loading. I just get the black banner with node red logo and a gray screen. I think it might be a caching issue as if I load it in chrome incognito mode It will load properly the first time. I also tried clearing my browser cache and it worked first time but not again after that.

I do use Duckdns if that makes a difference.

```
{
  "credential_secret": "#####",
  "dark_mode": false,
  "http_node": {
    "username": "####",
    "password": "##########"
  },
  "http_static": {
    "username": "#####",
    "password": "##########"
  },
  "ssl": true,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "require_ssl": true,
  "system_packages": [],
  "npm_packages": [],
  "init_commands": [],
  "log_level": "info",
  "users": [
    {
      "username": "######",
      "password": "############",
      "permissions": "*"
    }
  ],
  "port": 1880
}

```

I get this in my Node-red addon logs.....

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Hass.io Add-on: Node-RED
 Flow-based programming for the Internet of Things
-----------------------------------------------------------
 Add-on version: 3.0.1
 You are running the latest version of this add-on.
 System: HassOS 2.11  (armv7 / raspberrypi3)
 Home Assistant version: 0.91.4
 Supervisor version: 160
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to INFO
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] node-red.sh: executing... 
[20:01:22] INFO: Password is NOT in the Have I Been Pwned database! Nice!
[20:01:24] INFO: Password is NOT in the Have I Been Pwned database! Nice!
npm WARN node-red-project@0.0.1 No repository field.
npm WARN node-red-project@0.0.1 No license field.
audited 5 packages in 3.006s
found 0 vulnerabilities
[cont-init.d] node-red.sh: exited 0.
[cont-init.d] user.sh: executing... 
[cont-init.d] user.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[20:01:31] INFO: Starting Node-RED...
> hassio-addon-node-red@ start /opt
> node $NODE_OPTIONS node_modules/node-red/red.js "--settings" "/etc/node-red/config.js"
16 Apr 20:01:37 - [info] 
Welcome to Node-RED
===================
16 Apr 20:01:37 - [info] Node-RED version: v0.20.5
16 Apr 20:01:37 - [info] Node.js  version: v10.14.2
16 Apr 20:01:37 - [info] Linux 4.14.98-v7 arm LE
16 Apr 20:01:38 - [info] Loading palette nodes
16 Apr 20:01:45 - [info] Settings file  : /etc/node-red/config.js
16 Apr 20:01:45 - [info] Context store  : 'default' [module=memory]
16 Apr 20:01:45 - [info] User directory : /config/node-red/
16 Apr 20:01:45 - [info] Server now running at http://127.0.0.1:46836/
16 Apr 20:01:45 - [info] Active project : NodeRedHomeAssistant
16 Apr 20:01:45 - [info] Flows file     : /config/node-red/projects/NodeRedHomeAssistant/flows.json
[20:01:45] INFO: Starting NGinx...
16 Apr 20:01:45 - [info] Starting flows
16 Apr 20:01:46 - [info] Started flows
16 Apr 20:01:47 - [info] [mqtt-broker:Hassio MQTT] Connected to broker: mqtt://192.168.0.40:1883
16 Apr 20:01:50 - [info] [server:Home Assistant (working)] WebSocket Connected to http://hassio/homeassistant
[16/Apr/2019:20:02:02 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/bootstrap/css/bootstrap.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/jquery/css/smoothness/jquery-ui-1.10.3.custom.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/vendor.css HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /red/images/node-red.png HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/font-awesome/css/font-awesome.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /red/style.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /red/images/spin.svg HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/vendor.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/jsonata/jsonata.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/ace/ext-language_tools.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:03 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /vendor/ace/ace.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:04 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /red/main.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:20:02:04 +0100] 304 37.191.118.84, 172.30.32.1(172.30.32.2) GET /red/red.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
```

the logs after [16/Apr/2019:20:02:02 +0100] happen each time I try and load the ui which doenst load.

**Comment by bboser** [comment]
same problem. Node red starts fine, gets and processes events from hass, but won't show the gui to edit flows.

**Comment by mikenabhan** [comment]
i am having the same issue as @SurfingSpider.

I am able to access the UI on https://HASS_IP_ADDR:1880 after updating the port in the network config.  However, when I access through ingress it says "Lost connection to server, reconnecting..."

When i try to continue using my reverse proxy, only the top bar of the UI loads.

**Comment by mikenabhan** [comment]
I am able to confirm this fixes the issue with the UI not rending properly behind my reverse proxy.

https://github.com/hassio-addons/addon-node-red/issues/190#issuecomment-483463573

This does not resolve the issue of the addon timing out via ingress.

**Comment by SurfingSpider** [comment]
> I am able to confirm this fixes the issue with the UI not rending properly behind my reverse proxy.
> 
> [#190 (comment)](https://github.com/hassio-addons/addon-node-red/issues/190#issuecomment-483463573)
> 
> This does not resolve the issue of the addon timing out via ingress.
> 
> I will edit to say that I am using firefox.

After clearing cache I could also login if I opened the port up using the Network section of the add-on. I still have issues with ingress though.

**Comment by Den-W1** [comment]
Node Red 3.0.1 has same issues.
I was able to use the Network port to access the web ui  however ingress still is not working.  I get the black bar the the node red logo and a blank page as described above.  

Here is a partial Node Red log
> [16/Apr/2019:17:01:59 -0700] 200 -(192.168.1.116) GET /vendor/ace/snippets/javascript.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:01:59 -0700] 200 -(192.168.1.116) GET /vendor/ace/worker-javascript.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:01:59 -0700] 304 -(192.168.1.116) GET /vendor/ace/worker-javascript.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/ HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/socket.io/socket.io.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/css/app.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/icon120x120.png HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/js/app.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/i18n.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/css/app.min.less HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 200 -(192.168.1.116) GET /ui/socket.io/?EIO=3&transport=polling&t=MeecQm6 HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 200 -(192.168.1.116) GET /ui/socket.io/?EIO=3&transport=polling&t=MeecQmH&sid=SWooQ5Xc8OBTcWrTAAAA HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 304 -(192.168.1.116) GET /ui/icon64x64.png HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:02:01 -0700] 200 -(192.168.1.116) GET /ui/socket.io/?EIO=3&transport=polling&t=MeecQmo&sid=SWooQ5Xc8OBTcWrTAAAA HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/bootstrap/css/bootstrap.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/vendor.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/vendor.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/font-awesome/css/font-awesome.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/jquery/css/smoothness/jquery-ui-1.10.3.custom.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /red/images/spin.svg HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /red/style.min.css HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/ace/ace.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /red/images/node-red.png HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/ace/ext-language_tools.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /vendor/jsonata/jsonata.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /red/red.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[16/Apr/2019:17:04:47 -0700] 304 192.168.1.116, 172.30.32.1(172.30.32.2) GET /red/main.min.js HTTP/1.1 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)

**Comment by dreed47** [comment]
not sure why this was closed as the problem still persists with 3.0.1 and HA 0.91.4

**Comment by SurfingSpider** [comment]
Agreed don't think it should have been closed
